lib/v2/screens/explore_screens/invite/interactor/mappers/seeds_amount_change_mapper.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:seeds/v2/blocs/rates/viewmodels/rates_state.dart';
+import 'package:seeds/v2/datasource/local/settings_storage.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/explore_screens/invite/interactor/viewmodels/invite_state.dart';
+import 'package:seeds/v2/utils/rate_states_extensions.dart';
+
+class SeedsAmountChangeMapper extends StateMapper {
+  InviteState mapResultToState(InviteState currentState, RatesState rateState, String quantity) {
+    double parsedQuantity = double.tryParse(quantity) ?? 0;
+    double currentAvailable = currentState.availableBalance?.quantity ?? 0;
+    String alertMessage = _hasAlertMessage(currentAvailable, parsedQuantity);
```

Comment:
_hasAlertMessage sounds like a yes or not (true or false), but alertMessage is `alertMessage` 
A better name is _handleAlertMessage()

Reply:
also, i noticed in the state  String? alertMessage, is nullable, but this function defaults to ' '
Should it be null when there is not error?

Reply:
The only place in the state where is nullable is in the:
```
copyWith(
...
    String? alertMessage,
...
) 

```
I made it non-nullable to get the benefit of checking for the `isEmpty` and `isNotEmpty` property but I can still change it to nullable and make alertMessage == null or! = null.

If you like it changed it to nullable.

Reply:
I see, we should be using nullable instead of "" to represent if its set or not. 

We made an extension function called isNullOrEmpty 